src/main/java/org/kiwiproject/jersey/client/ServiceIdentifier.java

```diff
@@ -0,0 +1,49 @@
+package org.kiwiproject.jersey.client;
+
+import static org.kiwiproject.base.KiwiPreconditions.checkArgumentNotBlank;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.With;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Service definition used for the client connection
+ */
+@Getter
```

Comment:
Try adding `@Value` and see if it works (the all-args ctor generated by the builder _might_ cause issues)